Vungle/VungleAdapter/ALVungleMediationAdapter.m

```diff
@@ -618,262 +434,363 @@ + (MAAdapterError *)toMaxError:(nullable NSError *)vungleError
+{
+    MAAdapterError *adapterError = [ALVungleMediationAdapter toMaxError: withError];
+    [self.parentAdapter log: @"Interstitial failed to load with error: %@", adapterError];
+    [self.interstitialAdDelegate didFailToLoadInterstitialAdWithError: adapterError];
+}
+
+- (void)interstitialAdDidPresent:(VungleInterstitial * _Nonnull)interstitial
+{
+    [self.interstitialAdDelegate didDisplayInterstitialAd];
+}
+
+- (void)interstitialAdDidFailToPresent:(VungleInterstitial * _Nonnull)interstitial withError:(NSError * _Nonnull)withError
+{
+    MAAdapterError *adapterError = [MAAdapterError errorWithCode: -4205 errorString: @"Ad Display Failed" mediatedNetworkErrorCode: withError.code mediatedNetworkErrorMessage: withError.localizedDescription];;
+    [self.parentAdapter log: @"Interstitial ad failed to display with error: %@", adapterError];
+    [self.interstitialAdDelegate didFailToDisplayInterstitialAdWithError: adapterError];
+}
+
+- (void)interstitialAdDidClose:(VungleInterstitial * _Nonnull)interstitial
+{
+    [self.interstitialAdDelegate didHideInterstitialAd];
+}
+
+- (void)interstitialAdDidClick:(VungleInterstitial * _Nonnull)interstitial
+{
+    [self.interstitialAdDelegate didClickInterstitialAd];
 }
 
 @end
 
-@implementation ALVungleMediationAdapterRouter
+@implementation ALVungleMediationRewardedAdapterRouter
 
-#pragma mark - GDPR
+- (instancetype)initVungleRewardedAdDelegate:(id<MARewardedAdapterDelegate>)rewardedAdDelegate parentAdapter:(ALVungleMediationAdapter *)parentAdapter
+{
+    self = [super init];
+    if ( self )
+    {
+        self.rewardedAdDelegate = rewardedAdDelegate;
+        self.parentAdapter = parentAdapter;
+    }
+    return self;
+}
 
-- (void)updateUserPrivacySettingsForParameters:(id<MAAdapterParameters>)parameters consentDialogState:(ALConsentDialogState)consentDialogState
+#pragma mark - VungleRewardedDelegate
+
+- (void)rewardedAdDidLoad:(VungleRewarded * _Nonnull)rewarded
 {
-    if ( consentDialogState == ALConsentDialogStateApplies )
+    [self.rewardedAdDelegate didLoadRewardedAd];
+}
+
+- (void)rewardedAdDidFailToLoad:(VungleRewarded * _Nonnull)rewarded withError:(NSError * _Nonnull)withError
+{
+    MAAdapterError *adapterError = [ALVungleMediationAdapter toMaxError: withError];
+    [self.parentAdapter log: @"Rewarded failed to load with error: %@", adapterError];
+    [self.rewardedAdDelegate didFailToLoadRewardedAdWithError: adapterError];
+}
+
+- (void)rewardedAdDidPresent:(VungleRewarded * _Nonnull)rewarded
+{
+    [self.rewardedAdDelegate didDisplayRewardedAd];
+    [self.rewardedAdDelegate didStartRewardedAdVideo];
+}
+
+- (void)rewardedAdDidFailToPresent:(VungleRewarded * _Nonnull)rewarded withError:(NSError * _Nonnull)withError
+{
+    MAAdapterError *adapterError = [MAAdapterError errorWithCode: -4205 errorString: @"Ad Display Failed" mediatedNetworkErrorCode: withError.code mediatedNetworkErrorMessage: withError.localizedDescription];
```

Comment:
I saw `-4205` is used three times, might we define it as constant? Thanks a lot. :)